From d801ba05c2dcfa5291568e16d5fcf9b1abd74066 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 17 Nov 2016 12:31:29 +0200
Subject: [PATCH 0583/1345] fix: pinctl: cp110: Add missing SDIO functions to
 pin control driver

commit  41df468c642b073536ef8cab2ad7d9b626b04b6d from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add missing MPP[61:56] function 14 (SDIO) to cp110 pin control driver

Change-Id: I0684334305770691bf396119e48b41c8c429dfc5
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33877
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pinctrl/mvebu/pinctrl-armada-cp110.c |   18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c b/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
index 3fb2c8b..fd3dcb2 100644
--- a/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
+++ b/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
@@ -555,7 +555,8 @@ enum {
 		 MPP_VAR_FUNCTION(5,	"au",		"i2sdo_spdifo",	V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(6,	"spi0",		"clk",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(7,	"uart1",	"rxd",		V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(9,	"sata1",	"present_act",	V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(9,	"sata1",	"present_act",	V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"clk",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(57,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(2,	"mss_i2c",	"sda",		V_ARMADA_80X0_CP0_PLUS),
@@ -564,7 +565,8 @@ enum {
 		 MPP_VAR_FUNCTION(5,	"au",		"i2sbclk",	V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(6,	"spi0",		"mosi",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(7,	"uart1",	"txd",		V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(9,	"sata0",	"present_act",	V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(9,	"sata0",	"present_act",	V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"cmd",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(58,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(2,	"mss_i2c",	"sck",		V_ARMADA_80X0_CP0_PLUS),
@@ -573,7 +575,8 @@ enum {
 		 MPP_VAR_FUNCTION(5,	"au",		"i2sdi",	V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(6,	"spi0",		"miso",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(7,	"uart1",	"cts",		V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(8,	"led",		"clk",		V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(8,	"led",		"clk",		V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"d0",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(59,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(1,	"mss_gpio7",	NULL,		V_ARMADA_80X0_CP0_PLUS),
@@ -583,7 +586,8 @@ enum {
 		 MPP_VAR_FUNCTION(6,	"spi0",		"csn0",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(7,	"uart0",	"cts",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(8,	"led",		"stb",		V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(9,	"uart1",	"txd",		V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(9,	"uart1",	"txd",		V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"d1",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(60,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(1,	"mss_gpio6",	NULL,		V_ARMADA_80X0_CP0_PLUS),
@@ -593,7 +597,8 @@ enum {
 		 MPP_VAR_FUNCTION(6,	"spi0",		"csn1",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(7,	"uart0",	"rts",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(8,	"led",		"data",		V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(9,	"uart1",	"rxd",		V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(9,	"uart1",	"rxd",		V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"d2",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(61,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(1,	"mss_gpio5",	NULL,		V_ARMADA_80X0_CP0_PLUS),
@@ -604,7 +609,8 @@ enum {
 		 MPP_VAR_FUNCTION(7,	"uart0",	"txd",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(8,	"uart2",	"txd",		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(9,	"sata1",	"present_act",	V_ARMADA_80X0_CP0_PLUS),
-		 MPP_VAR_FUNCTION(10,	"ge",		"mdio",		V_ARMADA_80X0_CP0_PLUS)),
+		 MPP_VAR_FUNCTION(10,	"ge",		"mdio",		V_ARMADA_80X0_CP0_PLUS),
+		 MPP_VAR_FUNCTION(14,	"sdio",		"d3",		V_ARMADA_80X0_CP0_PLUS)),
 	MPP_MODE(62,
 		 MPP_VAR_FUNCTION(0,	"gpio",		NULL,		V_ARMADA_80X0_CP0_PLUS),
 		 MPP_VAR_FUNCTION(1,	"mss_gpio4",	NULL,		V_ARMADA_80X0_CP0_PLUS),
-- 
1.7.9.5

